overlord/snapstate/storehelpers.go

```diff
@@ -211,9 +211,9 @@ var installSize = func(st *state.State, snaps []minimalInstallInfo, userID int)
 	return total, nil
 }
 
-func setActionValidationSetsAndRequiredRevision(action *store.SnapAction, valsets []string, requiredRevision snap.Revision) {
+func setActionValidationSetsAndRequiredRevision(action *store.SnapAction, valsets []snapasserts.ValidationSetKey, requiredRevision snap.Revision) {
```

Comment:
I wonder if there are no circular import issues, whether it would make sense to also change the type inside SnapAction? it could be done as a separate PR

Reply:
I think it works but since I have to implement a UnmarshalJSON (with more tests), I'll leave that as a follow-up